File: rust/client/src/components/shared/IconBtn/IconBtn.js
import React, {memo} from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import Icon from '../Icon';
import classes from './IconBtn.css';

IconBtn.propTypes = {
  className: PropTypes.string,
  icon: PropTypes.string.isRequired,
  title: PropTypes.string,
  width: PropTypes.number,
  onClick: PropTypes.func.isRequired
};

function IconBtn({className, icon, title, width, onClick}) {
  return (
    <Icon
      className={cn(classes.root, className)}
      icon={icon}
      title={title}
      width={width}
      onClick={onClick}
    />
  );
}

export default memo(IconBtn);

File: rust+yew/client/src/components/shared/icon_btn/icon_btn.rs
use crate::components::shared::{Icon, IconType};
use yew::prelude::*;

#[derive(Properties, PartialEq)]
pub struct Props {
    #[prop_or_default]
    pub class: Classes,
    pub icon: IconType,
    #[prop_or_default]
    pub title: String,
    #[prop_or(20)]
    pub width: i32,
    pub on_click: Callback<MouseEvent>,
}

#[function_component(IconBtn)]
pub fn icon_btn(props: &Props) -> Html {
    log::trace!("render IconBtn");
    let css = css_mod::get!("icon_btn.css");
    html! {
        <Icon
            class={classes!(css["root"], props.class.clone())}
            icon={props.icon}
            title={props.title.clone()}
            width={props.width}
            on_click={props.on_click.clone()}
        />
    }
}
